caldav: Handle gracefully components containing no VEVENT's

dtzWill opened the pull request:
Currently such a resource causes an assertion failure,
effectively preventing use with such calendars entirely.

With this change, focal "gracefully" handles these by
ignoring them and moving along.

dtzWill committed:
caldav: Handle gracefully components containing no VEVENT's

Currently such data causes an assertion failure,
effectively preventing use with such calendars entirely.

With this change, focal gracefully handles these by
ignoring them and moving along.

ohwgiles commented on src/caldav-calendar.c:
```diff
@@ -396,7 +396,7 @@ static Event* create_event_from_parsed_xml(CaldavCalendar* cal, CaldavEntry* cde
 	icalcomponent* comp = icalparser_parse_string(cde->caldata);
 	icalcomponent* vev = icalcomponent_get_first_component(comp, ICAL_VEVENT_COMPONENT);
-	g_assert_nonnull(vev);
+	if (!vev) return NULL;
```
nitpick: focal [code style](https://github.com/ohwgiles/focal/blob/master/CONTRIBUTING.md#code-style) would put this on a second line. Good idea to run `clang-format` before pushing

ohwgiles commented on src/caldav-calendar.c:
```diff
@@ -505,11 +505,13 @@ static void sync_multiget_report_done(CURL* curl, CURLcode ret, void* user)
+				continue;
+			}
 		}
+		caldav_entry_free(cde);
```
Maybe it's just personal preference, but I think this can be horizontally squished a bit and made a little clearer without `continue` (I'm totally aware that I used `goto` in this file though...)

I'll push a little tweak

ohwgiles committed:
minor style tweaks

ohwgiles commented:
Hey, this is a cool and useful fix, thanks! Style tweak look ok to you? (I did not craft such a calendar to test it on)

ohwgiles commented:
As a side note, at some point this area of code could use a cleanup. It would be nice to unify `create_event_from_parsed_xml` and `caldav_entry_free` if possible and then remove the necessity to check which one to use... but there are totally bigger fish to fry for now

dtzWill commented:
On Thu, 24 Oct 2019 11:08:28 -0700, Oliver Giles <notifications@github.com> wrote:
> Hey, this is a cool and useful fix, thanks! Style tweak look ok to you? (I did not craft such a calendar to test it on)

Yes, I very much prefer the version after your changes.

And +1 for clang-format, will use it for future submissions for sure!
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/ohwgiles/focal/pull/81#issuecomment-546036628Non-text part: text/html

ohwgiles commented on src/caldav-calendar.c:
```diff
@@ -396,7 +396,8 @@ static Event* create_event_from_parsed_xml(CaldavCalendar* cal, CaldavEntry* cde
 	icalcomponent* vev = icalcomponent_get_first_component(comp, ICAL_VEVENT_COMPONENT);
-	g_assert_nonnull(vev);
+	if (!vev)
+		return NULL;
```
Just realised this will leak `comp` (`vev` not leaked as `icalcomponent_get_first_component` doesn't pass ownership). `icalcomponent_free` should be called before returning

dtzWill replied:
Okay, thanks for catching this! Easy fix, pushed the change on top of rebasing on latest master.

dtzWill committed:
caldav-calendar: fix leak of 'comp' in parse failure

Fix credit to review comment:

https://github.com/ohwgiles/focal/pull/81#discussion_r338911468

ohwgiles committed:
remove extra semicolon
